### ISSUE TITLE
Name collision for /version endpoint

### ISSUE THREAD

**Issue by bluenote10** [acacode/swagger-typescript-api#41]
Another small issue I encountered: It is currently not possible to have a `/version` endpoint, because of a name collision with the internal `version` variable. For example:

```yaml
---
openapi: 3.0.0
info:
  title: SomeAPI
paths:
  "/version":
    get:
      responses:
        '200':
          description: 
          content:
            application/json:
              schema:
                type: string
```

Relevant parts of generated code:

```ts
export class Api<SecurityDataType> {
  public baseUrl = "";
  public title = "SomeAPI";
  public version = "";

  // and later

  version = {
    /**
     * @name versionList
     * @request GET:/version
     */
    versionList: (params?: RequestParams) => this.request<string, any>(`/version`, "GET", params, null),
  };
```

Error:

```
Duplicate identifier 'version'.ts(2300)
```

**Comment by js2me** [comment]
Yes, any changes of source code for solving this issue will be major.  
As a solution I think needs to move out baseUrl, title, version properties from Api class

**Comment by js2me** [comment]
@bluenote10 I think `title` and `version` are not needed for development. May be it should be in description Api class ? (comments)

**Comment by js2me** [comment]
All other not route properties probably can be contains in inherited Http class

**Comment by js2me** [comment]
Should be fixed in release 1.7.0

**Comment by bluenote10** [comment]
Works perfectly, thanks!
